### PR TITLE
Upgrade ci to go v1.23.3

### PR DESCRIPTION
**Pull request by Ex4amp1e**
Issue: https://github.com/networkservicemesh/deployments-k8s/issues/12294

### CODE REVIEW

File: .ci/yamllint.yml [hunk 1]
```diff
@@ -9,3 +9,5 @@ rules:
   truthy: disable
   # 80 chars should be enough, but don't fail if a line is longer
   line-length: disable
+  comments:
+    min-spaces-from-content: 1
```
Comment (denis-tingaikin) [inline .ci/yamllint.yml:13]: why do we need this change?

**Review by denis-tingaikin — COMMENTED** [review]
can we also update v1.23.1 to v1.23.2?

File: .ci/yamllint.yml (hunk 1 above)
Comment (Ex4amp1e) [reply .ci/yamllint.yml:13]: To fix sdk yamllint issue https://github.com/networkservicemesh/sdk/actions/runs/10899858521/job/30246211877?pr=1664 `too few spaces before comment  (comments)`. We have discussed, that this linter config update is ok.

Comment (denis-tingaikin) [reply .ci/yamllint.yml:13]: What exactly line(change) causes  the problem in https://github.com/networkservicemesh/sdk/pull/1664?

Comment (Ex4amp1e) [reply .ci/yamllint.yml:13]: ./.golangci.yml
  60:14     warning  too few spaces before comment  (comments)
  61:14     warning  too few spaces before comment  (comments)
  62:14     warning  too few spaces before comment  (comments)

      - G301 # Fixable
      - G302 # Fixable
      - G306 # Fixable

**Comment by denis-tingaikin** [comment]
v1.23.3 is released. Can we update go-version?

**Review by denis-tingaikin — APPROVED** [review]

**Pull request merged**